docs/pages/inboxes/send-messages.md

```diff
@@ -36,6 +36,183 @@ try await group.send(content: "Hello everyone")
 
 :::
 
+## Optimistically send messages
+
+When a user sends a message with XMTP, they might experience a slight delay between sending the message and seeing their sent message display in their app UI.
+
+Typically, the slight delay is caused by the app needing to wait for the XMTP network to finish processing the message before the app can display the message in its UI.
+
+Messaging without optimistic sending:
+
+![Messaging without optimistic sending. Note the slight delay after clicking Send.](https://raw.githubusercontent.com/xmtp/docs-xmtp-org/main/docs/pages/img/without-opt-sending.gif)
+
+Note the slight delay after clicking **Send**.
+
+Implement optimistic sending to immediately display the sent message in the sender’s UI while processing the message in the background. This provides the user with immediate feedback and enables them to continue messaging without waiting for their previous message to finish processing.
+
+Messaging with optimistic sending:
+
+![Messaging with optimistic sending. The message displays immediately for the sender, with a checkmark indicator displaying once the message has been successfully sent.](https://raw.githubusercontent.com/xmtp/docs-xmtp-org/main/docs/pages/img/with-opt-sending.gif)
+
+The message displays immediately for the sender, with a checkmark indicator displaying once the message has been successfully sent.
+
+### Key UX considerations
+
+- After initially sending a message optimistically, show the user an indicator that the message is still being processed. After successfully sending the message, show the user a success indicator.
+
+- If an optimistically sent message fails to send, give the user an option to retry sending the message or cancel sending. Use a try/catch block to intercept errors and allow the user to retry or cancel.
+
+### 1. Optimistically send a message locally
+
+There are two steps to optimistically send a message:
+
+1. Send the message locally so it can display immediately in the sender's UI.
+2. Send the message to the XMTP network so it can be delivered to the recipient.
+
+:::code-group
+
+```tsx [Browser]
+// Send a message optimistically (displays immediately in the sender's UI)
```

Comment:
i think "displaying in the UI" may not be the best phrasing here, since it's on the dev to do that. i think a better explanation is that the message is added to the local database, so when you query for messages, it will be there. i need to write some more tests to confirm that the optimistically sent message appear in message streams.

messages have a delivery status, and optimistically sent messages initially have an "unpublished" status. this status can be used to indicate in a UI whether or not it's been published on the network.